chaosazure/machine/actions.py

```diff
@@ -117,6 +118,58 @@ def start_machine(filter: str = None,
     __start_stopped_machines(client, stopped_machines)
 
 
+def stress_cpu(filter: str = None,
+               duration: int = 120,
+               configuration: Configuration = None,
+               secrets: Secrets = None):
+    """
+    Stress CPU up to 100% at random machine.
+
+    Parameters
+    ----------
+    filter : str
+        Filter the virtual machines. If the filter is omitted all machines in
+        the subscription will be selected as potential chaos candidates.
+        Filtering example:
+        'where resourceGroup=="myresourcegroup" and name="myresourcename"'
+    """
+    logger.info(
+        "Start stress_cpu: configuration='{}', filter='{}'".format(
+            configuration, filter))
+
+    choice = __fetch_machine_at_random(filter, configuration, secrets)
+
+    logger.info("Stressing CPUs machine: {}".format(choice['name']))
+    client = __init_client(secrets, configuration)
+
+    os_type = __get_os_type(choice)
+    if os_type == OS_WINDOWS:
+        command_id = 'RunPowerShellScript'
+        script_name = "cpu_stress_test.ps1"
+    elif os_type == OS_LINUX:
+        command_id = 'RunShellScript'
+        script_name = "cpu_stress_test.sh"
+    else:
+        raise Exception("Cannot run CPU stress test on OS: %s" % os_type)
+
+    with open(os.path.join(os.path.dirname(__file__),
+                           "scripts", script_name)) as file:
+        script_content = file.read()
+
+    parameters = {
+        'command_id': command_id,
+        'script': [script_content],
+        'parameters': [
+            {'name': "duration", 'value': duration}
+        ]
+    }
+
+    poller = client.virtual_machines.run_command(
+        choice['resourceGroup'], choice['name'], parameters)
+    result = poller.result()  # Blocking till executed
```

Comment:
Can it take a timeout? Even though you passed a duration to the script :)

Reply:
Seems feasible will take a look

Reply:
Thanks. It's really more of a safeguard than anything else.

Reply:
It make sense, we do not want it to hang. 

The question would be how big this timeout should be? I cannot take the original duration as executing 'run_command' takes longer than executing script only. It has to be sth like "test duration time"+"submit and results retrieval duration". I will make some tests and based on the observation will define some value required for this additional time.
If you have better idea let me know.

Reply:
@mkaszub That sounds good: duration time + an epsilon time

Reply:
Good point. Also, I suggest you make sure to pass the duration as a parameter of the action with a default sensible value.